util/OptionsDB.h

```diff
@@ -309,6 +310,10 @@ class FO_COMMON_API OptionsDB
       * the command line */
     void        SetFromCommandLine(const std::vector<std::string>& args);
 
+    /** if the xml file exists and is not an outdated version, load the values
+     *    of the DB options into XMLDoc, then fill the DB using SetFromXML */
+    void        SetFromFile(const boost::filesystem::path& file_path, const std::string& version = "");
```

Comment:
That was exactly it. (I copied the next lines for SetFromXML, then reworded it.)